src/Containers.jsx

```diff
@@ -27,112 +28,262 @@ import { PodActions } from './PodActions.jsx';
+                      onClick={() => setCommitModal(true)}>
+            {_("Commit")}
+        </DropdownItem>
+    );
+    actions.push(<DropdownSeparator key="separator-2" />);
+    actions.push(
+        <DropdownItem key="delete"
+                      className="pf-m-danger"
+                      onClick={deleteContainer}>
+            {_("Delete")}
+        </DropdownItem>
+    );
+
+    const kebab = (
+        <Dropdown toggle={<KebabToggle onToggle={isOpen => setActionsKebabOpen(isOpen)} />}
+                  isOpen={isActionsKebabOpen}
+                  isPlain
+                  position="right"
+                  dropdownItems={actions} />
+    );
+
+    const containerDeleteModal =
+        <ContainerDeleteModal
+            containerWillDelete={container}
+            handleCancelContainerDeleteModal={() => setDeleteModal(false)}
+            handleRemoveContainer={handleRemoveContainer}
+        />;
+    const containerCheckpointModal =
+        <ContainerCheckpointModal
+            handleCheckpointContainer={handleCheckpointContainer}
+            handleCheckpointContainerDeleteModal={() => setCheckpointModal(false)}
+            containerWillCheckpoint={container}
+            checkpointInProgress={checkpointInProgress}
+        />;
+    const containerRestoreModal =
+        <ContainerRestoreModal
+            handleRestoreContainer={handleRestoreContainer}
+            handleRestoreContainerDeleteModal={() => setRestoreModal(false)}
+            containerWillCheckpoint={container}
+            restoreInProgress={restoreInProgress}
+        />;
+    const containerRemoveErrorModal =
+        <ForceRemoveModal
+            name={container.Names}
+            handleCancel={() => setRemoveErrorModal(false)}
+            handleForceRemove={handleForceRemoveContainer}
+            reason={_("Container is currently running.")}
+        />;
+
+    const containerCommitModal =
+        <ContainerCommitModal
+            onHide={() => setCommitModal(false)}
+            container={container}
+            version={version}
+        />;
+
+    return (
+        <>
+            {kebab}
+            {deleteModal && containerDeleteModal}
+            {checkpointModal && containerCheckpointModal}
+            {restoreModal && containerRestoreModal}
+            {removeErrorModal && containerRemoveErrorModal}
+            {commitModal && containerCommitModal}
+        </>
+    );
+};
+
+class Containers extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            width: 0,
+        };
+        this.renderRow = this.renderRow.bind(this);
+        this.onWindowResize = this.onWindowResize.bind(this);
+
+        this.cardRef = React.createRef();
 
-    restoreContainer(container) {
-        this.setState({
-            containerWillRestore: container,
-            selectContainerRestoreModal: true,
-        });
+        window.addEventListener('resize', this.onWindowResize);
```

Comment:
should there be debouncing for resizing events?

Reply:
That should be triggered only once on page load as it's in the component constructor.